lib/services/credentials/credential.service.js

```diff
@@ -292,7 +293,7 @@ function validateNewCredentialScopes (credentialConfig, credentialDetails) {
 // This function validates all user defined properties, excluding scopes
 function validateNewCredentialProperties (credentialConfig, credentialDetail) {
   // Tmp — horrible hack to remove.
-  const credentialDetails = JSON.parse(JSON.stringify(credentialDetail));
+  const credentialDetails = clone(credentialDetail);
```

Comment:
Are there deep properties? Why not object.assing?

Reply:
Unfortunately yes, it has got deep properties and I am forced to make a clone of this object because functions here and there modify this object and make the whole flow break.

I hope I can get rid of this once we'll refactor the entire credential service, but I am afraid we will have to live with this for a while.

Reply:
actually, the strange part is that `validate` modifies some object.  

if this is a default behavior of `ajv.validate` then this clone should probably be inside lib/schemas generic method. unlikely that this is the only place that has a complex structure to validate  

In any case, are you sure that it is really required to modify the object?

Reply:
The reason why `ajv.validate` is modifying the object is because:

1. It puts default values on properties that haven't been provided
2. Performs coercing (automatic conversion) on the passed object.

So it's something that we effectively want.

Now to be honest I do not really remember why I had to put such a bad hack in the code (it's completely my fault, I should have commented it better; I'll make sure to do that next time), but I recall I was spending a lot of time on it.

If that's ok with you, I'd skip this for now and open a separate issue; otherwise I could get stuck trying to remember the details and delaying this pull request a lot.

Does that make sense to you?